lib/widgets/search/main_search_bar.dart

```diff
@@ -110,16 +114,27 @@ class PositionedFloatingSearchBar extends SearchDelegate<String> {
 
   @override
   Widget buildSuggestions(BuildContext context) {
+    BuildingSingleton buildingSingleton = BuildingSingleton();
+    List<RoomCoordinate> roomList = <RoomCoordinate>[];
+    for (Building building in buildingSingleton.buildings) {
+      building.floors.forEach((floorName, floor) => floor
+          .coordinatesByGivenTypes({"ROOM"}).forEach(
+              (room) => roomList.add(room)));
+    }
```

Comment:
I would move the logic of getting list of rooms to the building_singleton as this logic might be repeated many times in different widgets.

Reply:
Done!